### PR TITLE
change cache path from fixed to configurable parameter

### LINKED ISSUE #343: [feature request]metadata cache path can be configured
**What would you like to be added**:
The k8s's resource metadata are saved to the fixed path `/etc/kubernetes/cache/`.  I think that it's better to change it according to bootstrap parameter.

**Why is this needed**:
Maybe some user want to save metadata to specified path. 

**others**
/kind feature

### PR DESCRIPTION
**Pull request by luckymrwang**
/kind feature

#### Which issue(s) this PR fixes:

Fixes #343

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

### CODE REVIEW

**Comment by Fei-Guo** [comment]
please fix the go-lint error.

**Comment by luckymrwang** [comment]
> please fix the go-lint error.

@Fei-Guo I have fixed the go-lint error.

File: pkg/yurthub/storage/disk/storage.go [hunk 1]
```diff
@@ -30,8 +30,7 @@ import (
 )
 
 const (
-	cacheBaseDir = "/etc/kubernetes/cache/"
```
Comment (rambohe-ch) [inline pkg/yurthub/storage/disk/storage.go:33]: I think we need keep the `cacheBaseDir` as default path for disk storage. if input parameter `dir` is empty, so we can use the default path.

File: pkg/yurthub/storage/disk/storage.go [hunk 2]
```diff
@@ -42,9 +41,6 @@ type diskStorage struct {
 
 // NewDiskStorage creates a storage.Store for caching data into local disk
 func NewDiskStorage(dir string) (storage.Store, error) {
-	if dir == "" {
```
Comment (rambohe-ch) [inline pkg/yurthub/storage/disk/storage.go:45]: It's need to check the validity of input parameter. so please do not delete it.

Comment (Fei-Guo) [reply pkg/yurthub/storage/disk/storage.go:45]: Actually, nil check is not enough for validity if we allow user input directory. We need to check more such as the directory is at least available.

File: pkg/yurthub/storage/disk/storage.go (hunk 1 above)
Comment (luckymrwang) [reply pkg/yurthub/storage/disk/storage.go:33]: `DiskCachePath` has been defined the default path `/etc/kubernetes/cache` to backward compatibility in the `cmd/yurthub/app/options/options.go`.

Comment (luckymrwang) [reply pkg/yurthub/storage/disk/storage.go:33]: > I think we need keep the `cacheBaseDir` as default path for disk storage. if input parameter `dir` is empty, so we can use the default path.

`DiskCachePath` has been defined default path `/etc/kubernetes/cache/` to `backward compatibility` in the `cmd/yurthub/app/options/options.go`.

File: pkg/yurthub/storage/disk/storage.go (hunk 2 above)
Comment (luckymrwang) [reply pkg/yurthub/storage/disk/storage.go:45]: > Actually, nil check is not enough for validity if we allow user input directory. We need to check more such as the directory is at least available.

@Fei-Guo The following code can cover it what the `dir` is empty or not valid path.

**Comment by Fei-Guo** [comment]
I think it is ok for now that if people explicitly specify the "" path then we fail it.
/lgtm

File: pkg/yurthub/storage/disk/storage.go (hunk 2 above)
Comment (luckymrwang) [reply pkg/yurthub/storage/disk/storage.go:45]: > It's need to check the validity of input parameter. so please do not delete it.

I have fixed it

**Comment by luckymrwang** [comment]
> I think it is ok for now that if people explicitly specify the "" path then we fail it.
> /lgtm

I changed the previous code which juded `dir` @rambohe-ch referred to. Please review it again.

**Pull request merged**